src/PasswordChangeDialog/PasswordChangeDialog.test.js

```diff
@@ -33,28 +33,26 @@ describe("PasswordChangeDialog", () => {
   it("returns form information to callback when successfully validated", async () => {
     const onSubmit = jest.fn(data => data);
     const elements = setup({ onSubmit });
-    await act(async () => {
-      userEvent.type(elements.inputs.currentPassword, "abc123");
-      userEvent.type(elements.inputs.newPassword, "coffee podium dvdplayer");
-      userEvent.type(
-        elements.inputs.newPasswordRepeat,
-        "coffee podium dvdplayer"
-      );
-      fireEvent.submit(elements.submit);
-    });
-    expect(onSubmit).toHaveReturnedWith({
-      currentPassword: "abc123",
-      newPassword: "coffee podium dvdplayer",
-      newPasswordRepeat: "coffee podium dvdplayer"
-    });
+    userEvent.type(elements.inputs.currentPassword, "abc123");
+    userEvent.type(elements.inputs.newPassword, "coffee podium dvdplayer");
+    userEvent.type(
+      elements.inputs.newPasswordRepeat,
+      "coffee podium dvdplayer"
+    );
+    fireEvent.submit(elements.submit);
+    await waitFor(() =>
+      expect(onSubmit).toHaveReturnedWith({
+        currentPassword: "abc123",
+        newPassword: "coffee podium dvdplayer",
+        newPasswordRepeat: "coffee podium dvdplayer"
+      })
+    );
   });
   it("doesnt call callback on validation errors", async () => {
     const onSubmit = jest.fn();
     const elements = setup({ onSubmit });
-    await act(async () => {
-      userEvent.type(elements.inputs.newPassword, "abc123"); // top 100 password
-      fireEvent.submit(elements.submit);
-    });
-    expect(onSubmit).not.toHaveBeenCalled();
+    userEvent.type(elements.inputs.newPassword, "abc123"); // top 100 password
+    fireEvent.submit(elements.submit);
+    await waitFor(() => expect(onSubmit).not.toHaveBeenCalled());
```

Comment:
Test should be testing for error message on screen rather than lack of callback execution.